server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java

```diff
@@ -179,16 +179,24 @@ public void testSortAndPaginateWithInProgress() throws Exception {
             inProgressSnapshots.add(startFullSnapshot(repoName, snapshotName));
         }
         awaitNumberOfSnapshotsInProgress(inProgressCount);
-        awaitClusterState(
-            state -> state.custom(SnapshotsInProgress.TYPE, SnapshotsInProgress.EMPTY)
-                .entries()
-                .stream()
-                .flatMap(s -> s.shards().stream())
-                .allMatch(
-                    e -> e.getKey().getIndexName().equals("test-index-1") == false
-                        || e.getValue().state() == SnapshotsInProgress.ShardState.SUCCESS
-                )
-        );
+        try {
+            ClusterServiceUtils.awaitClusterState(logger,
+                state -> state.custom(SnapshotsInProgress.TYPE, SnapshotsInProgress.EMPTY)
+                    .entries()
+                    .stream()
+                    .flatMap(s -> s.shards().stream())
+                    .allMatch(
+                        e -> e.getKey().getIndexName().equals("test-index-1") == false
+                            || e.getValue().state() == SnapshotsInProgress.ShardState.SUCCESS
+                    ),
+                internalCluster().getInstance(ClusterService.class, internalCluster().getMasterName()),
+                3 // Short time-out is enough, we don't want wait 30 sec if there's no updates
+            );
+        } catch (Throwable ignore) {
+            // Happens when the test-index-1 snapshots weren't all finished successfully, but the snapshots are blocked on the data nodes
```

Comment:
I don't think we can do this. A. we create a test that runs/sleeps for 30s randomly and B. why even have the assertion/wait if it doesn't actually work in all cases to begin with.
We should instead find a way to more deterministically block the io in the repo or have a cleaner state await predicate I think.

Reply:
Agreed, that's an ugly solution. I've been trying to find a way to deterministically coordinate the process, but haven't been successful yet. I will give it another go.